src/test/cloudWatchLogs/changeLogSearch.test.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+import { CloudWatchLogsSettings, createURIFromArgs } from '../../cloudWatchLogs/cloudWatchLogsUtils'
+import {
+    LogStreamRegistry,
+    ActiveTab,
+    CloudWatchLogsData,
+    CloudWatchLogsResponse,
+} from '../../cloudWatchLogs/registry/logStreamRegistry'
+import { Settings } from '../../shared/settings'
+
+describe('changeLogSearch', async function () {
+    let testRegistry: LogStreamRegistry
+    const config = new Settings(vscode.ConfigurationTarget.Workspace)
+
+    const fakeGetLogEvents = async (): Promise<CloudWatchLogsResponse> => {
+        return {
+            events: [
+                {
+                    message: 'we just got some log events!',
+                },
+            ],
+        }
+    }
+
+    const oldComponenents = {
+        logGroupInfo: {
+            groupName: 'this-is-a-group',
+            regionName: 'this-is-a-region',
+        },
+        parameters: { streamName: 'this-is-a-stream', filterPattern: 'this is a bad filter!' },
+    }
+    const newText = 'this is a good filter!'
+    const newComponents = {
+        ...oldComponenents,
+        parameters: {
+            ...oldComponenents,
+            filterPattern: newText,
+        },
+    }
+
+    const oldData: CloudWatchLogsData = {
+        data: [
+            {
+                message: 'Here is some original text that we want to overwrite.',
+            },
+        ],
+        parameters: oldComponenents.parameters,
+        logGroupInfo: oldComponenents.logGroupInfo,
+        retrieveLogsFunction: fakeGetLogEvents,
+        busy: false,
+    }
+
+    const newData: CloudWatchLogsData = {
```

Comment:
Can this test data be re-used across tests?
Reducing code duplication in tests is desirable too, just not to the same degree as other logic.

Reply:
I tried moving most of the functions and data that were used to `utils.test.ts`. Let me know if there is a better place or more that I could move.